### PR TITLE
Point users to logs when running 'mlflow ui' fails

### PR DESCRIPTION
**Pull request by mikehuston**
The change attempts to address #91 by pointing users to the logs of
trying to start the mlflow server instead of printing an exception
that does not explain the cause for failure and is slightly noisy.

Here are the logs when the server is run and the desired port is being used by another process:

**Before change**
```
(env) Michaels-MacBook-Pro:mlflow mhuston$ mlflow ui
[2018-07-08 16:36:08 -0700] [17612] [INFO] Starting gunicorn 19.9.0
[2018-07-08 16:36:08 -0700] [17612] [ERROR] Connection in use: ('127.0.0.1', 5000)
[2018-07-08 16:36:08 -0700] [17612] [ERROR] Retrying in 1 second.
[2018-07-08 16:36:09 -0700] [17612] [ERROR] Connection in use: ('127.0.0.1', 5000)
[2018-07-08 16:36:09 -0700] [17612] [ERROR] Retrying in 1 second.
[2018-07-08 16:36:10 -0700] [17612] [ERROR] Connection in use: ('127.0.0.1', 5000)
[2018-07-08 16:36:10 -0700] [17612] [ERROR] Retrying in 1 second.
[2018-07-08 16:36:11 -0700] [17612] [ERROR] Connection in use: ('127.0.0.1', 5000)
[2018-07-08 16:36:11 -0700] [17612] [ERROR] Retrying in 1 second.
[2018-07-08 16:36:12 -0700] [17612] [ERROR] Connection in use: ('127.0.0.1', 5000)
[2018-07-08 16:36:12 -0700] [17612] [ERROR] Retrying in 1 second.
[2018-07-08 16:36:13 -0700] [17612] [ERROR] Can't connect to ('127.0.0.1', 5000)
Traceback (most recent call last):
  File "/Users/mhuston/Code/mlflow/env/bin/mlflow", line 11, in <module>
    load_entry_point('mlflow', 'console_scripts', 'mlflow')()
  File "/Users/mhuston/Code/mlflow/env/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/mhuston/Code/mlflow/env/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/mhuston/Code/mlflow/env/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/mhuston/Code/mlflow/env/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/mhuston/Code/mlflow/env/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/mhuston/Code/mlflow/mlflow/cli.py", line 131, in ui
    mlflow.server._run_server(file_store, file_store, host, port, 1)
  File "/Users/mhuston/Code/mlflow/mlflow/server/__init__.py", line 48, in _run_server
    env=env_map, stream_output=True)
  File "/Users/mhuston/Code/mlflow/mlflow/utils/process.py", line 38, in exec_cmd
    raise ShellCommandException("Non-zero exitcode: %s" % (exit_code))
mlflow.utils.process.ShellCommandException: Non-zero exitcode: 1
```

**After change**
```
(env) Michaels-MacBook-Pro:mlflow mhuston$ mlflow ui
[2018-07-08 18:17:31 -0700] [20335] [INFO] Starting gunicorn 19.9.0
[2018-07-08 18:17:31 -0700] [20335] [ERROR] Connection in use: ('127.0.0.1', 5000)
[2018-07-08 18:17:31 -0700] [20335] [ERROR] Retrying in 1 second.
[2018-07-08 18:17:32 -0700] [20335] [ERROR] Connection in use: ('127.0.0.1', 5000)
[2018-07-08 18:17:32 -0700] [20335] [ERROR] Retrying in 1 second.
[2018-07-08 18:17:33 -0700] [20335] [ERROR] Connection in use: ('127.0.0.1', 5000)
[2018-07-08 18:17:33 -0700] [20335] [ERROR] Retrying in 1 second.
[2018-07-08 18:17:34 -0700] [20335] [ERROR] Connection in use: ('127.0.0.1', 5000)
[2018-07-08 18:17:34 -0700] [20335] [ERROR] Retrying in 1 second.
[2018-07-08 18:17:35 -0700] [20335] [ERROR] Connection in use: ('127.0.0.1', 5000)
[2018-07-08 18:17:35 -0700] [20335] [ERROR] Retrying in 1 second.
[2018-07-08 18:17:36 -0700] [20335] [ERROR] Can't connect to ('127.0.0.1', 5000)
Running the mlflow server failed. Please see the logs above for details.
```

### CODE REVIEW

File: mlflow/cli.py [hunk 1]
```diff
@@ -128,7 +129,12 @@ def ui(file_store, host, port):
     The UI will be visible at http://localhost:5000 by default.
     """
     # TODO: We eventually want to disable the write path in this version of the server.
-    mlflow.server._run_server(file_store, file_store, host, port, 1)
+    try:
+        mlflow.server._run_server(file_store, file_store, host, port, 1)
+    except ShellCommandException:
+        print("There was an issue running the mlflow server. Please see the logs above for details.",
```
Comment (mikehuston) [inline mlflow/cli.py]: I also considered making the exception message more specific than `"Non-zero exitcode: 1` and printing the exception message instead, but I felt what I came up with was still not as user-friendly as the text here.

**Comment by mateiz** [comment]
Unfortunately removing the stack trace might hinder debugging. The specific error message "Non-zero exitcode" might not be great, but could there be other exceptions where you do want to see the stack trace? Maybe we should print a nicer message but also keep the stack trace.

**Comment by mateiz** [comment]
I see, that makes sense. Since it's only for ShellCommandException and only in this CLI handler, I think it's actually fine then. Can you also add it to the `mlflow server` command to be consistent?

**Comment by mikehuston** [comment]
Sounds good, thanks @mateiz. I also added it to the `mlflow server` command.

**Pull request merged**